### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by kousthubraja**
- weighting scheme change for committee selection
- Changed reward logic
- Trust score to self shouldn't be allowed

### CODE REVIEW

File: app/codes/committeemanager.py [hunk 1]
```diff
@@ -28,22 +29,30 @@ def get_number_from_hash(block_hash):
...
-    previous_idx = 0
-    while len(selections) < k:
-        random.seed(seed_prefix + previous_idx)
-        choice = random.choices(population, weights=weights)[0]
-        index = population.index(choice)
-        previous_idx = index
-        selections.append(choice)
-        del population[index]
-        del weights[index]
+    v = [random.random() ** (1 / w) for w in weights]
```
Comment (swapnilpawar31415) [inline app/codes/committeemanager.py:36]: Shouldn't this be using the seed_prefix?

File: app/codes/state_updater.py [hunk 2]
```diff
@@ -181,8 +181,8 @@ def add_block_reward(cur, creator, blockindex):
     if creator is None:
         return False
     reward = 0
-    RATIO = 2 / 3
-    STARTING_REWARD = 1000
+    RATIO = 0.99
+    STARTING_REWARD = 100
     block_step = math.ceil(float(blockindex) / 1000000)
     reward = STARTING_REWARD * pow(RATIO, (block_step - 1)) * pow(10, NEWRL_TOKEN_DECIMAL)
```
Comment (swapnilpawar31415) [inline app/codes/state_updater.py:187]: this should be int() or math.floor()

File: app/codes/transactionmanager.py [hunk 3]
```diff
@@ -438,6 +438,9 @@ def econvalidator(self):
             if not wallet1valid or not wallet2valid:
                 print("One of the wallets is invalid")
                 self.validity = 0
+            elif wallet1 == wallet2:
```
Comment (swapnilpawar31415) [inline app/codes/transactionmanager.py:441]: the comparison should be of pids ..
since one pid can have multiple wallet, this check is not sufficient.

Comment (kousthubraja) [reply app/codes/transactionmanager.py:441]: yes. haven't thought of that. was only checking single wallet score.

File: app/codes/state_updater.py (hunk 2 above)
Comment (kousthubraja) [reply app/codes/state_updater.py:187]: you mean 
`reward = math.floor(STARTING_REWARD * pow(RATIO, (block_step - 1)) * pow(10, NEWRL_TOKEN_DECIMAL))` ?

File: app/codes/committeemanager.py (hunk 1 above)
Comment (kousthubraja) [reply app/codes/committeemanager.py:36]: yes, that was missed. correcting.

**Pull request merged**